src/util/dstring.h

```diff
@@ -198,4 +200,13 @@ struct hash<dstringt> // NOLINT(readability/identifiers)
 };
 }
 
+template <>
+struct diagnostics_helpert<dstringt>
+{
+  static std::string diagnostics_as_string(const dstringt &dstring)
```

Comment:
I'm considering making this a function that takes a `std::ostream&` and writes some string representation of its argument there rather than returning string. Thoughts?

Reply:
We can leave this for future work I'd say :)